application/Filters/DebugToolbar.php

```diff
@@ -42,16 +42,19 @@ public function after(RequestInterface $request, ResponseInterface $response)
 			$toolbar = Services::toolbar(new App());
 			$stats   = $app->getPerformanceStats();
 
-			return $response->appendBody(
-				$toolbar->run(
-					$stats['startTime'],
-					$stats['totalTime'],
-					$stats['startMemory'],
-					$request,
-					$response
-				)
-			);
-		}
+                        $output = $toolbar->run(
+                                $stats['startTime'], $stats['totalTime'], $stats['startMemory'], $request, $response
+                        );
+
+                        helper('filesystem');
+                        helper('text');
+                        $fileName = 'debugbar_' . random_string();
+                        write_file('./' . $fileName, $output, 'w+');
+                        return $response->appendBody(PHP_EOL .
+                                '<iframe src="/getdebugbar.php?f=' . $fileName .
```

Comment:
... and GET a custom Route.

The Ajax version was working nice.

Reply:
Yes,  the ajax version had almost full functionality but the source code of the toolbar still ended up in the DOM appended to the application code... And kept showing in the source-view of the browser...  And that's what I was trying to fix... On the other hand, the iframe achieves what I want but it sandboxes the code to its interior...

Reply:
The toolbar code didn't end up in the view-source of the browser. I could find it in the "developper's tools" 
So, no more iframe to load the toolbar. Back to AJAX. Take a look to the new commits I've made. It's now fully functional and only ads one <script> tag to the end of the application generated code to load the toolbar.